File: 算法/代码随想录算法PDF/01数组/283.移动零.py
#
# @lc app=leetcode.cn id=283 lang=python3
#
# [283] 移动零
#

# @lc code=start
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        slowIndex = 0
        for fastIndex in range(len(nums)):
            if (nums[fastIndex] != 0):
                nums[slowIndex] = nums[fastIndex]
                slowIndex += 1
        while slowIndex < len(nums):
            nums[slowIndex] = 0
            slowIndex += 1
# @lc code=end

File: 算法/代码随想录算法PDF/00调试源代码/283.移动零.cpp
/*
 * @lc app=leetcode.cn id=283 lang=cpp
 *
 * [283] 移动零
 */

// @lc code=start
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int slowIndex = 0; // 慢指针
        for (int fastIndex = 0; fastIndex < nums.size(); fastIndex++) {
            if (nums[fastIndex] != 0) {
                nums[slowIndex++] = nums[fastIndex];  // 替换非0元素
            }
        }
        for (;slowIndex < nums.size(); slowIndex++) {
            nums[slowIndex] = 0; // 末尾补0
        }
    }
};
// @lc code=end

